src/main/java/htsjdk/samtools/SamPairUtil.java

```diff
@@ -246,12 +246,14 @@ else if (rec1.getReadUnmappedFlag() && rec2.getReadUnmappedFlag()) {
             mapped.setMateUnmappedFlag(true);
             // For the mapped read, set it's mateCigar to null, since the other read must be unmapped
             mapped.setAttribute(SAMTag.MC.name(), null);
+            mapped.setAttribute(SAMTag.MQ.name(), null);

```

Comment:
move to before setting the `MC` since that's the order elsewhere: it makes it easier to find.